### PR TITLE
Add `EntityType`s and `EntityTypeRegistry`.

### PR DESCRIPTION
**Pull request by williambl**
Please codereview this to death, I'm not convinced I've done everything right.

Example usage:

```java
Id id = new Id("mymod:myentity")
Registries.ENTITY_TYPE.register(id, a -> EntityTypes.createEntityType(MyEntity.class, id))
```

### CODE REVIEW

**Comment by valoeghese** [comment]
ok how do I make checkstyle run on this

**Comment by valoeghese** [comment]
also It's 1am I'm not reviewing rn

**Comment by williambl** [comment]
I have no clue how checkstyle works... you should probably sleep lol.

**Comment by valoeghese** [comment]
push a commit

**Comment by williambl** [comment]
Okay, I'll bump the version number.

**Comment by valoeghese** [comment]
and now you have to deal with the checkstyle errors we already had as well as the new ones

**Comment by valoeghese** [comment]
have fun fixing my bad code :Yeefrickinghaw:

**Comment by williambl** [comment]
:irritatered:

**Comment by valoeghese** [comment]
Run `.\gradlew check` or `.\gradlew checkstyleMain` to get info locally

**Comment by williambl** [comment]
Ah okay, thanks.

**Comment by williambl** [comment]
Oh wait, I think I know why the test failed - gradlew isn't marked as executable.

**Comment by williambl** [comment]
Okay, now checkstyle actually runs. Only issue now is that my copy of IDEA has suddenly decided it no longer wants to run, so I can't actually fix the style issues.

**Comment by williambl** [comment]
Checkstyle stuff all fixed now.

File: src/main/java/io/github/minecraftcursedlegacy/mixin/MixinEntityRegistry.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityRegistry;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Accessor;
+import org.spongepowered.asm.mixin.gen.Invoker;
+
+import java.util.Map;
+
+@Mixin(EntityRegistry.class)
+public interface MixinEntityRegistry {
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/mixin/MixinEntityRegistry.java]: we put accessors in a different mixin json with a different naming scheme

**Review by valoeghese — CHANGES_REQUESTED** [review]
api.accessors.json

File: src/main/java/io/github/minecraftcursedlegacy/mixin/MixinEntityRegistry.java (hunk 1 above)
Comment (valoeghese) [reply src/main/java/io/github/minecraftcursedlegacy/mixin/MixinEntityRegistry.java]: and in a different package I think

Comment (williambl) [reply src/main/java/io/github/minecraftcursedlegacy/mixin/MixinEntityRegistry.java]: Done.

File: src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorEntityRegistry.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		throw new AssertionError("mixin");
+	}
+
+	@Accessor("STRING_ID_TO_CLASS")
+	static Map<String, Class<? extends Entity>> getSTRING_ID_TO_CLASS() {
+		throw new AssertionError("mixin");
+	}
+
+	@Accessor("STRING_ID_TO_CLASS")
+	static void setSTRING_ID_TO_CLASS(Map<String, Class<? extends Entity>> value) {
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorEntityRegistry.java]: can you make these method names look a bit nicer perhaps? Specifying `STRING_ID_TO_CLASS` as your accessor value means you can name it whatever you want (as long as it doesn't conflict with an existing method).

**Review by valoeghese — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorEntityRegistry.java (hunk 2 above)
Comment (valoeghese) [reply src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorEntityRegistry.java]: also applies to the other lowerCamelCaseMIXED_WITH_UPPER_SNAKE_CASE methods

Comment (williambl) [reply src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorEntityRegistry.java]: Renamed.

**Review by valoeghese — APPROVED** [review]

**Pull request merged**